File: platform/sdk-examples/typescript/sdk-structure-demo.ts
#!/usr/bin/env node
/**
 * Respan SDK Structure Demo
 *
 * This example shows the SDK structure and how to initialize clients
 * without making actual API calls.
 *
 * Usage:
 *     npm run build && node dist/examples/sdk-structure-demo.js
 *     or
 *     npx tsx examples/sdk-structure-demo.ts
 */

import { config } from 'dotenv';
import {
  DatasetAPI,
  ExperimentAPI,
  EvaluatorAPI,
  LogAPI,
} from '../src/index.js';

// Load environment variables with override
config({ override: true });

function main() {
  console.log('🏗️  Respan SDK Structure Demo');
  console.log('='.repeat(50));

  // 1. Show SDK structure
  console.log('📦 Available API Clients:');
  console.log('   📊 DatasetAPI - Manage datasets');
  console.log('   🧪 ExperimentAPI - Manage experiments');
  console.log('   🔍 EvaluatorAPI - Work with evaluators');
  console.log('   📝 LogAPI - Manage logs');
  console.log();

  // 2. Show initialization
  console.log('🔧 Client Initialization:');
  const apiKey = process.env.RESPAN_API_KEY || 'your-api-key-here';
  const baseUrl = process.env.RESPAN_BASE_URL || 'http://localhost:8000';

  console.log(`   🔑 API Key: ${apiKey.length > 10 ? apiKey.slice(0, 10) + '...' : apiKey}`);
  console.log(`   🌐 Base URL: ${baseUrl}`);
  console.log();

  // Initialize clients (no API calls yet)
  console.log('🚀 Initializing clients...');
  const datasetApi = new DatasetAPI({ apiKey, baseUrl });
  const experimentApi = new ExperimentAPI({ apiKey, baseUrl });
  const evaluatorApi = new EvaluatorAPI({ apiKey, baseUrl });
  const logApi = new LogAPI({ apiKey, baseUrl });

  console.log('   ✅ DatasetAPI initialized');
  console.log('   ✅ ExperimentAPI initialized');
  console.log('   ✅ EvaluatorAPI initialized');
  console.log('   ✅ LogAPI initialized');
  console.log();

  // 3. Show available methods
  console.log('📋 Available Dataset Methods:');
  const datasetMethods = Object.getOwnPropertyNames(Object.getPrototypeOf(datasetApi))
    .filter(method => !method.startsWith('_') && typeof datasetApi[method as keyof typeof datasetApi] === 'function')
    .sort();
  
  for (const method of datasetMethods) {
    console.log(`   • ${method}()`);
  }
  console.log();

  console.log('📋 Available Evaluator Methods:');
  const evaluatorMethods = Object.getOwnPropertyNames(Object.getPrototypeOf(evaluatorApi))
    .filter(method => !method.startsWith('_') && typeof evaluatorApi[method as keyof typeof evaluatorApi] === 'function')
    .sort();
  
  for (const method of evaluatorMethods) {
    console.log(`   • ${method}()`);
  }
  console.log();

  // 4. Show data types
  console.log('📝 Available Data Types:');
  console.log('   📊 Dataset Types:');
  console.log('      • DatasetCreate - For creating new datasets');
  console.log('      • DatasetUpdate - For updating existing datasets');
  console.log('      • LogManagementRequest - For managing logs in datasets');
  console.log();
  console.log('   🧪 Experiment Types:');
  console.log('      • ExperimentCreate - For creating new experiments');
  console.log('      • ExperimentUpdate - For updating existing experiments');
  console.log('      • ExperimentColumnType - For experiment columns');
  console.log('      • ExperimentRowType - For experiment rows');
  console.log();
  console.log('   🔍 Evaluator Types:');
  console.log('      • Evaluator - Individual evaluator model');
  console.log('      • EvaluatorList - List of evaluators with pagination');
  console.log();
  console.log('   📝 Log Types:');
  console.log('      • RespanLogParams - For creating logs');
  console.log('      • RespanFullLogParams - Full log data');
  console.log('      • LogList - List of logs with pagination');
  console.log();

  // 5. Usage examples
  console.log('💡 Basic Usage Examples:');
  console.log();
  console.log('   // Create a dataset');
  console.log('   const dataset = await datasetApi.create({');
  console.log('     name: "My Dataset",');
  console.log('     description: "Test dataset",');
  console.log('     type: "sampling"');
  console.log('   });');
  console.log();
  console.log('   // List evaluators');
  console.log('   const evaluators = await evaluatorApi.list();');
  console.log();
  console.log('   // Create a log');
  console.log('   const log = await logApi.create({');
  console.log('     model: "gpt-4",');
  console.log('     input: "Hello world",');
  console.log('     output: "Hi there!"');
  console.log('   });');
  console.log();

  console.log('🎯 Next Steps:');
  console.log('   1. Set RESPAN_API_KEY environment variable');
  console.log('   2. Run examples: npm run example:simple-evaluator');
  console.log('   3. Check out the dataset-workflow example');
  console.log('   4. Explore the experiment-workflow example');
  console.log();
  console.log('✅ SDK structure demo completed!');
}

if (import.meta.url === `file://${process.argv[1]}`) {
  main();
}

File: platform/sdk-examples/python/sdk_structure_demo.py
#!/usr/bin/env python3
"""
Respan SDK Structure Demo

This example shows the SDK structure and how to initialize clients
without making actual API calls.

Usage:
    python examples/sdk_structure_demo.py
"""

import os
from dotenv import load_dotenv

load_dotenv()

from respan.datasets.api import DatasetAPI, SyncDatasetAPI
from respan.evaluators.api import EvaluatorAPI, SyncEvaluatorAPI
from respan.prompts.api import PromptAPI, SyncPromptAPI
from respan_sdk.respan_types.dataset_types import DatasetCreate, DatasetUpdate, LogManagementRequest
from respan.types.evaluator_types import Evaluator, EvaluatorList
from respan_sdk.respan_types.prompt_types import Prompt, PromptVersion


def main():
    """Demo SDK structure and initialization"""
    
    print("🏗️  Respan SDK Structure Demo")
    print("=" * 50)
    
    # 1. Show SDK structure
    print("📦 Available API Clients:")
    print("   📊 DatasetAPI (async) - Manage datasets")
    print("   📊 SyncDatasetAPI (sync) - Manage datasets synchronously") 
    print("   🔍 EvaluatorAPI (async) - Work with evaluators")
    print("   🔍 SyncEvaluatorAPI (sync) - Work with evaluators synchronously")
    print("   📝 PromptAPI (async) - Manage prompts and versions")
    print("   📝 SyncPromptAPI (sync) - Manage prompts and versions synchronously")
    print()
    
    # 2. Show initialization
    print("🔧 Client Initialization:")
    api_key = os.getenv("RESPAN_API_KEY", "your-api-key-here")
    base_url = os.getenv("RESPAN_BASE_URL", "http://localhost:8000")
    
    print(f"   🔑 API Key: {api_key[:10]}..." if len(api_key) > 10 else f"   🔑 API Key: {api_key}")
    print(f"   🌐 Base URL: {base_url}")
    print()
    
    # Initialize clients (no API calls yet)
    print("🚀 Initializing clients...")
    dataset_api = DatasetAPI(api_key=api_key, base_url=base_url)
    sync_dataset_api = SyncDatasetAPI(api_key=api_key, base_url=base_url)
    evaluator_api = EvaluatorAPI(api_key=api_key, base_url=base_url)
    sync_evaluator_api = SyncEvaluatorAPI(api_key=api_key, base_url=base_url)
    prompt_api = PromptAPI(api_key=api_key, base_url=base_url)
    sync_prompt_api = SyncPromptAPI(api_key=api_key, base_url=base_url)
    
    print("   ✅ DatasetAPI (async) initialized")
    print("   ✅ SyncDatasetAPI initialized")  
    print("   ✅ EvaluatorAPI (async) initialized")
    print("   ✅ SyncEvaluatorAPI initialized")
    print("   ✅ PromptAPI (async) initialized")
    print("   ✅ SyncPromptAPI initialized")
    print()
    
    # 3. Show available methods
    print("📋 Available Dataset Methods:")
    dataset_methods = [method for method in dir(dataset_api) if not method.startswith('_') and callable(getattr(dataset_api, method))]
    for method in sorted(dataset_methods):
        print(f"   • {method}()")
    print()
    
    print("📋 Available Evaluator Methods:")
    evaluator_methods = [method for method in dir(evaluator_api) if not method.startswith('_') and callable(getattr(evaluator_api, method))]
    for method in sorted(evaluator_methods):
        print(f"   • {method}()")
    print()
    
    # 4. Show available prompt methods
    print("📋 Available Prompt Methods:")
    prompt_methods = [method for method in dir(prompt_api) if not method.startswith('_') and callable(getattr(prompt_api, method))]
    for method in prompt_methods[:8]:  # Show first 8 methods
        print(f"      • {method}")
    print(f"      ... and {len(prompt_methods) - 8} more methods")
    print()

    # 5. Show data types
    print("📝 Available Data Types:")
    print("   📊 Dataset Types:")
    print("      • DatasetCreate - For creating new datasets")
    print("      • DatasetUpdate - For updating existing datasets")
    print("      • LogManagementRequest - For managing logs in datasets")
    print()
    print("   🔍 Evaluator Types:")
    print("      • Evaluator - Individual evaluator model")
    print("      • EvaluatorList - List of evaluators with pagination")
    print()
    print("   📝 Prompt Types:")
    print("      • Prompt - Individual prompt model")
    print("      • PromptVersion - Prompt version model")
    print()
    
    # 5. Show example usage patterns (no API calls)
    print("💡 Usage Patterns:")
    print()
    print("   🔄 Async Pattern:")
    print("   ```python")
    print("   async def my_function():")
    print("       dataset_api = DatasetAPI(api_key='...', base_url='...')")
    print("       datasets = await dataset_api.list()")
    print("       return datasets")
    print("   ```")
    print()
    print("   🔄 Sync Pattern:")
    print("   ```python")
    print("   def my_function():")
    print("       dataset_api = SyncDatasetAPI(api_key='...', base_url='...')")
    print("       datasets = dataset_api.list()")
    print("       return datasets")
    print("   ```")
    print()
    
    # 6. Show workflow overview
    print("🔄 Typical Workflows:")
    print()
    print("   📊 Dataset Workflow:")
    print("      1. Initialize API client")
    print("      2. List available evaluators")
    print("      3. Create a dataset")
    print("      4. Add logs to dataset")
    print("      5. Run evaluation on dataset")
    print("      6. Get evaluation results")
    print()
    print("   📝 Prompt Workflow:")
    print("      1. Initialize PromptAPI client")
    print("      2. Create a new prompt")
    print("      3. Create versions with different configurations")
    print("      4. List and retrieve prompts and versions")
    print("      5. Update prompt and version properties")
    print()
    
    print("✅ SDK structure demo complete!")
    print()
    print("🚀 Next Steps:")
    print("   • Set up your .env file with RESPAN_API_KEY")
    print("   • Run: python examples/simple_evaluator_example.py")
    print("   • Run: python examples/dataset_workflow_example.py")
    print("   • Check: examples/README.md for more details")


if __name__ == "__main__":
    main()